--- vorbis.c	2003/11/10 02:18:43	1.1
+++ vorbis.c	2004/03/05 05:19:53	1.2
@@ -62,13 +62,13 @@
     void * sym;
     
     vorbis_opened = 0;
-    lib = dlopen("libvorbis.so", RTLD_GLOBAL | RTLD_LAZY);
+    lib = dlopen("libvorbis.so.0", RTLD_GLOBAL | RTLD_LAZY);
     if (!lib) {
-        vorbis_error = "Unable to open libvorbis.so";
+        vorbis_error = "Unable to open libvorbis.so.0";
         return vorbis_opened;
     }
 
-    lib = dlopen("libvorbisfile.so", RTLD_GLOBAL | RTLD_LAZY);
+    lib = dlopen("libvorbisfile.so.3", RTLD_GLOBAL | RTLD_LAZY);
     if (lib) {
         vorbis_opened = 1;
         

